### PR TITLE
Final Merge.  App ready for production

### PR DESCRIPTION
**Pull request by Metsanpeitto**
### Final Merge

:+1:  This is the final merge of the app ready for production.
It uses an API that returns weather data. This data is fetched to obtain the value of the different countries and cities.
[Presentation Video](https://metsanpeitto.github.io/Capstone-3_Redux/#/)
The app implements the following requirements : 

**Design** 
- Colors.
- Typography: font face, size, and weight.
- Layout: composition and space between elements.

**Technical Requirements**
- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).

**Interactions**
- Home page
  - When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, it displays some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

- Details page
  - On the details page, the web app retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**
-  Do unit tests for pure functions in Redux code.
-  Do integration tests using the React Testing Library.
  - Mock the access to the API so that tests don't send actual requests.
  - Mock the connection to the Redux Store.

### CODE REVIEW

**Review by Dipeshtwis — COMMENTED** [review]
## Status: Invalid ❌

## Hi @Metsanpeitto,

It's great to see that you have worked so hard on this project but I can not proceed. Please fix the issue mentioned below and then request for code review

## Reason

- [ ] Linter should be set up correctly for this repository. I think you have missed adding the Github workflows to your repository. Please set up and make sure all the linters are passing while requesting another review.


**Don't worry you still have same number of requests left as previous.**


Cheers and Happy Coding :clap: :clap: :clap: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

**_Dipesh Kumar_**

**Comment by Metsanpeitto** [comment]
:sweat_smile:  Sorry. I mistakenly add linters into the gitignore file. Thanks for pointing it out :+1:

File: src/api/api.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
+/* eslint-disable camelcase */
```
Comment (Dipeshtwis) [inline src/api/api.js]: - [ ] I would not recommend disabling eslint. instead, try to make the code standard to follow the eslint rules. :slightly_smiling_face:

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    if (this.props) {
+      // eslint-disable-next-line react/prop-types
```
Comment (Dipeshtwis) [inline src/App.js]: - [ ] Here as well, I would not recommend disabling eslint. instead, try to make the code standard to follow the eslint rules. :slightly_smiling_face:

File: README.md [hunk 3]
```diff
@@ -1 +1,141 @@
...
+
+Give a ⭐️ if you like this project!
+
+=======
+
+## Acknowledgments
+
+- Microverse for giving me this chance
+- The amazing code reviewers for making me improve every day :thumbsup:
+
```
Comment (Dipeshtwis) [inline README.md]: - [ ] The Acknowledgements section should give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Metsanpeitto, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Deployed the application using Github pages and working as expected.
:+1: The App used redux to store the data. Well Done.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/api/api.js (hunk 1 above)
Comment (Metsanpeitto) [reply src/api/api.js]: Ok. I found a way to solve it. Thanks for pointing it out :+1:

File: src/App.js (hunk 2 above)
Comment (Metsanpeitto) [reply src/App.js]: Ok. I solved it already. :+1:

File: README.md (hunk 3 above)
Comment (Metsanpeitto) [reply README.md]: Perfect! I have already fixed it. Thanks for your time :1st_place_medal:

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**